static/js/src/advantage/subscribe/checkout/components/UserInfoForm/UserInfoForm.tsx

```diff
@@ -0,0 +1,346 @@
+                  lineHeight: "24px",
+
+                  "::placeholder": {
+                    color: "#666",
+                  },
+                  ":-webkit-autofill": {
+                    color: "#666",
+                  },
+                },
+              },
+            }}
+            onFocus={() => {
+              setCardFieldFocus(true);
+            }}
+            onBlur={() => {
+              setCardFieldFocus(false);
+            }}
+            onChange={(e) => {
+              if (e.complete && !e.error) {
+                setCardValid(true);
+                setCardFieldError(null);
+              } else {
+                setCardValid(false);
+                if (e.error) {
+                  setCardFieldError(e.error);
+                }
+              }
+            }}
+          />
+        </div>
+      </FormRow>
+      <Field
+        as={Input}
+        type="text"
+        id="name"
+        name="name"
+        label="Name:"
+        stacked
+        validate={validateRequired}
+        error={touched?.name && errors?.name}
+      />
+      <FormRow label="I’m buying for:">
+        <div className="u-sv3 p-form p-form--inline" role="group">
+          <Field
+            as={RadioInput}
+            name="buyingFor"
+            value="myself"
+            label="Myself"
+            defaultChecked={values.buyingFor === "myself"}
+          />
+          <Field
+            as={RadioInput}
+            name="buyingFor"
+            value="organisation"
+            label="An organisation"
+            defaultChecked={values.buyingFor === "organisation"}
+          />
+        </div>
+      </FormRow>
+      <Field
+        as={Input}
+        type="text"
+        id="organisationName"
+        name="organisationName"
+        disabled={values.buyingFor === "myself"}
+        label="Organisation:"
+        stacked
+        validate={validateOrganisationName}
+        error={touched?.organisationName && errors?.organisationName}
+      />
+      <Field
+        as={Input}
+        type="text"
+        id="address"
+        name="address"
+        label="Address:"
+        stacked
+        validate={validateRequired}
+        error={touched?.address && errors?.address}
+      />
+      <Field
+        as={Input}
+        type="text"
+        id="postalCode"
+        name="postalCode"
+        label="Postal code:"
+        stacked
+        validate={validateRequired}
+        error={touched?.postalCode && errors?.postalCode}
+      />
+      <Field
+        as={Input}
+        type="text"
+        id="city"
+        name="city"
+        label="City:"
+        stacked
+        validate={validateRequired}
+        error={touched?.city && errors?.city}
+      />
+    </>
+  );
+
+  return (
+    <Row>
+      {isEditing ? editMode : displayMode}
+      {window.accountId ? (
+        <div className="u-align--right">
+          <ActionButton onClick={toggleEditing} loading={isSubmitting}>
```

Comment:
Should we have a cancel button? When logged in and with payment info set if I click edit I loose the payment method details and there is no way to revert without refreshing.

Reply:
Added cancel button.